test/helpers/node.go

```diff
@@ -135,12 +135,18 @@ func (s *SSHMeta) ExecuteContext(ctx context.Context, cmd string, stdout io.Writ
 		Stdout: stdout,
 		Stderr: stderr,
 	}
-	err := s.sshClient.RunCommandContext(ctx, command)
+
+	ch := make(chan error, 1)
+
+	go func(errChan chan error) {
+		errChan <- s.sshClient.RunCommandContext(ctx, command)
+	}(ch)
+
 	select {
+	case err := <-ch:
```

Comment:
This is not that simple. If the context is canceled, we will never know if the ssh connection was actually terminated or not.

Reply:
Right now, we block our CI entirely if `RunCommandContext` gets stuck (which it definitely can). If the context is canceled, then we still try to terminate the session anyway in `RunCommandContext`. This is best effort anyway, as these attempts to do so are done in a goroutine.

If the context is canceled, we return an error to the caller, indicating that something is wrong. It's better to at least signal this and let the CI try to continue rather than blocking forever.

Reply:
@ianvernon I think this is almost good but I think we should call `RunCommandInBackground()` instead so the SSH session gets closed when we cancel the context.

The documentation for `RunCommandContext()` is quite misleading as I don't see how it can cancel an active session for which the command has not completed yet.